Have a docker image with a more flexible ENTRYPOINT/CMD

**Is your feature request related to a problem? Please describe.**
Some CI/CD tools cannot overwrite entrypoint for a container but try to run commands inside the containe rwith bash or sh. This makes it impossible to use the current Docker image ([same issue with the official terraform image](https://github.com/hashicorp/docker-hub-images/issues/37)). To be able to use it one has to build it's own image with tfsec. 

**Describe the solution you'd like**
I would like to have an official docker image without tfsec hardcoded into the entrypoint.

**Describe alternatives you've considered**
Build my own image which works but an extra step which could be avoided with a more flexible Docker image.

Status on this? I have faced similiar issues  

```sh
$ tfsec .

...

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x7f8c31]
goroutine 1 [running]:
main.glob..func2(0xc4e460, 0xc000134ab0, 0x1, 0x1)
	/src/cmd/tfsec/main.go:166 +0x751
github.com/spf13/cobra.(*Command).execute(0xc4e460, 0xc00009c030, 0x1, 0x1, 0xc4e460, 0xc00009c030)
	/src/vendor/github.com/spf13/cobra/command.go:846 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0xc4e460, 0x8bf70b, 0xc0000bdf78, 0x4059df)
	/src/vendor/github.com/spf13/cobra/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/src/vendor/github.com/spf13/cobra/command.go:887
main.main()
	/src/cmd/tfsec/main.go:57 +0x31
Cleaning up file based variables
00:00
ERROR: Job failed: exit code 2
```

pipeline config:

```yaml
tfsec_validate:
  stage: validate
  image: 
    name: liamg/tfsec:v0.37
    entrypoint:
      - "/usr/bin/env"
  only:
    - merge_requests
  script:
    - tfsec .
```

--- a/Dockerfile.ci
+++ b/Dockerfile.ci
@@ -1,0 +1,20 @@
+FROM golang:1.16-alpine AS build-env
+
+ARG tfsec_version=0.0.0
+
+COPY ./ /src
+WORKDIR /src
+ENV CGO_ENABLED=0
+RUN go build \
+  -a \
+  -ldflags "-X github.com/tfsec/tfsec/version.Version=${tfsec_version} -s -w -extldflags '-static'" \
+  -mod=vendor \
+  ./cmd/tfsec
+
+FROM alpine
+# use a non-privileged user
+USER nobody
+# work somewhere where we can write
+COPY --from=build-env /src/tfsec /usr/bin/tfsec
+# as we specified an entrypoint, this is appended as an argument (i.e., `tfsec --help`)
+CMD [ "tfsec", "--help" ]

--- a/README.md
+++ b/README.md
@@ -54,6 +54,15 @@ tfsec .
 
 As an alternative to installing and running tfsec on your system, you may run tfsec in a Docker container.
 
+There are a number of Docker options available
+
+| Image Name | Base | Comment |
+|------------|------|---------|
+|[tfsec/tfsec](https://hub.docker.com/repository/docker/tfsec/tfsec)|alpine|Normal tfsec image|
+|[tfsec/tfsec-alpine](https://hub.docker.com/repository/docker/tfsec/tfsec-alpine)|alpine|Exactly the same as tfsec/tfsec, but for those whole like to be explicit|
+|[tfsec/tfsec-ci](https://hub.docker.com/repository/docker/tfsec/tfsec-ci)|alpine|tfsec with no entrypoint - useful for CI builds where you want to override the command|
+|[tfsec/tfsec-scratch](https://hub.docker.com/repository/docker/tfsec/tfsec-scratch)|scratch|An image build on scratch - nothing frilly, just runs tfsec|
+
 To run:
 
 ```bash

--- a/scripts/publish-image.sh
+++ b/scripts/publish-image.sh
@@ -3,47 +3,34 @@
 set -e 
 
 IMAGES=(tfsec/tfsec)
+
+function publish_image() {
+    WORKING_IMAGE=$1
+    TARGET_IMAGE=$2
+
+    echo "publishing ${TARGET_IMAGE}..."
+    # push the patch tag - eg; v0.36.15
+    docker tag ${WORKING_IMAGE} ${TARGET_IMAGE}:${TRAVIS_TAG}
+    docker push ${TARGET_IMAGE}:${TRAVIS_TAG}
+
+    # push the minor tag - eg; v0.36
+    docker tag ${WORKING_IMAGE} ${TARGET_IMAGE}:${TRAVIS_TAG%.*}
+    docker push ${TARGET_IMAGE}:${TRAVIS_TAG%.*}
+
+    # push the latest tag
+    docker tag ${WORKING_IMAGE} ${TARGET_IMAGE}:latest
+    docker push ${TARGET_IMAGE}:latest
+
+}
+
 for IMAGE in "${IMAGES[@]}"; do
     echo "building ${IMAGE}..."
     docker build --build-arg tfsec_version="${TRAVIS_TAG}" -f Dockerfile -t "${IMAGE}" .
-    docker build --build-arg tfsec_version="${TRAVIS_TAG}" -f Dockerfile.scratch -t "${IMAGE}"-scratch .
+    docker build --build-arg tfsec_version="${TRAVIS_TAG}" -f Dockerfile.scratch -t "${IMAGE}-scratch" .
+    docker build --build-arg tfsec_version="${TRAVIS_TAG}" -f Dockerfile.ci -t "${IMAGE}-ci" .
 
-    echo "publishing ${IMAGE}..."
-    # push the patch tag - eg; v0.36.15
-    docker tag ${IMAGE} ${IMAGE}:${TRAVIS_TAG}
-    docker push ${IMAGE}:${TRAVIS_TAG}
-
-    # push the minor tag - eg; v0.36
-    docker tag ${IMAGE} ${IMAGE}:${TRAVIS_TAG%.*}
-    docker push ${IMAGE}:${TRAVIS_TAG%.*}
-
-    # push the latest tag
-    docker tag ${IMAGE} ${IMAGE}:latest
-    docker push ${IMAGE}:latest
-
-    echo "publishing ${IMAGE}-alpine..."
-    # push the patch tag - eg; v0.36.15
-    docker tag ${IMAGE} ${IMAGE}-alpine:${TRAVIS_TAG}
-    docker push ${IMAGE}-alpine:${TRAVIS_TAG}
-
-    # push the minor tag - eg; v0.36
-    docker tag ${IMAGE} ${IMAGE}-alpine:${TRAVIS_TAG%.*}
-    docker push ${IMAGE}-alpine:${TRAVIS_TAG%.*}
-
-    # push the latest tag
-    docker tag ${IMAGE} ${IMAGE}-alpine:latest
-    docker push ${IMAGE}-alpine:latest
-
-    echo "publishing ${IMAGE}-scratch..."
-    # push the patch tag - eg; v0.36.15
-    docker tag ${IMAGE}-scratch ${IMAGE}-scratch:${TRAVIS_TAG}
-    docker push ${IMAGE}-scratch:${TRAVIS_TAG}
-
-    # push the minor tag - eg; v0.36
-    docker tag ${IMAGE}-scratch ${IMAGE}-scratch:${TRAVIS_TAG%.*}
-    docker push ${IMAGE}-scratch:${TRAVIS_TAG%.*}
-
-    # push the latest tag
-    docker tag ${IMAGE}-scratch ${IMAGE}-scratch:latest
-    docker push ${IMAGE}-scratch:latest
+    publish_image "${IMAGE}" "${IMAGE}"
+    publish_image "${IMAGE}" "${IMAGE}-alpine"
+    publish_image "${IMAGE}-scratch" "${IMAGE}-scratch"
+    publish_image "${IMAGE}-ci" "${IMAGE}-ci"
 done;

--- a/scripts/publish-image.sh
+++ b/scripts/publish-image.sh
@@ -4,10 +4,10 @@ set -e
 
 IMAGES=(tfsec/tfsec)
 
-for IMAGE in ${IMAGES[@]}; do
+for IMAGE in "${IMAGES[@]}"; do
     echo "building ${IMAGE}..."
-    docker build --build-arg tfsec_version=${TRAVIS_TAG} -f Dockerfile -t ${IMAGE} .
-    docker build --build-arg tfsec_version=${TRAVIS_TAG} -f Dockerfile.scratch -t ${IMAGE}-scratch .
+    docker build --build-arg tfsec_version="${TRAVIS_TAG}" -f Dockerfile -t "${IMAGE}" .
+    docker build --build-arg tfsec_version="${TRAVIS_TAG}" -f Dockerfile.scratch -t "${IMAGE}"-scratch .
 
     echo "publishing ${IMAGE}..."
     # push the patch tag - eg; v0.36.15

--- a/scripts/publish-image.sh
+++ b/scripts/publish-image.sh
@@ -10,16 +10,16 @@ function publish_image() {
 
     echo "publishing ${TARGET_IMAGE}..."
     # push the patch tag - eg; v0.36.15
-    docker tag ${WORKING_IMAGE} ${TARGET_IMAGE}:${TRAVIS_TAG}
-    docker push ${TARGET_IMAGE}:${TRAVIS_TAG}
+    docker tag "${WORKING_IMAGE}" "${TARGET_IMAGE}":"${TRAVIS_TAG}"
+    docker push "${TARGET_IMAGE}":"${TRAVIS_TAG}"
 
     # push the minor tag - eg; v0.36
-    docker tag ${WORKING_IMAGE} ${TARGET_IMAGE}:${TRAVIS_TAG%.*}
-    docker push ${TARGET_IMAGE}:${TRAVIS_TAG%.*}
+    docker tag "${WORKING_IMAGE}" "${TARGET_IMAGE}":"${TRAVIS_TAG%.*}"
+    docker push "${TARGET_IMAGE}":"${TRAVIS_TAG%.*}"
 
     # push the latest tag
-    docker tag ${WORKING_IMAGE} ${TARGET_IMAGE}:latest
-    docker push ${TARGET_IMAGE}:latest
+    docker tag "${WORKING_IMAGE}" "${TARGET_IMAGE}":latest
+    docker push "${TARGET_IMAGE}":latest
 
 }
 

--- a/scripts/publish-image.sh
+++ b/scripts/publish-image.sh
@@ -3,7 +3,6 @@
 set -e 
 
 IMAGES=(tfsec/tfsec)
-
 for IMAGE in "${IMAGES[@]}"; do
     echo "building ${IMAGE}..."
     docker build --build-arg tfsec_version="${TRAVIS_TAG}" -f Dockerfile -t "${IMAGE}" .

